---
# Code from Zarifeh
title: "scRNA analysis"
date: "`r Sys.Date()`"
output: 
  html_document:
    toc: true 
    toc_depth: 1
    toc_float:
      collapsed: true
      smooth_scroll: true
    number_sections: false  
    theme: united  
    highlight: tango 
---

# Imports, paths and global variables

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE)

library(stringr)
library(dplyr)
library(Seurat)
library(patchwork)
library(ggplot2)
library(openxlsx)
library(harmony)
library(SingleR)

# input to seurat is filtered_feature_by_barcodes_matrix from cell ranger, which contains counts per feature
proj_path <- "/ocean/projects/cis240075p/asachan/datasets/fibroblast_senescence"
dataset_path <- file.path(proj_path, "seurat_all_treatments.Rds")
seurat_obj <- readRDS(dataset_path)
# cell_ranger_outs_path <- ""
# mtPattern: a regex pattern to match against features to look for mitochondrial DNA; may need to adjust for
# sample species---"^MT-" works for human samples and "^mt-" for mice
mtPattern <- "^mt-"
doCite <- FALSE
```

# Initialize seurat object using raw counts from cell_ranger output (if unprocessed)

```{r load}
# list of all objects per sample
seurats <- list()
for (s1 in 1:length(sample_names)) {
  # read 10x count matrix generated from fastq files using cell-ranger
  input.data <- Read10X(count_paths[s1])
  # the seurat object is initialized by the count matrix
  seurat_obj <- CreateSeuratObject(counts = input.data, project = sample_names[s1])
  # add additional assay (CITE-seq) if present
  if (doCite) {
    adt.data <- Read10X(adt_count_paths[s1], gene.column = 1)
    colnames(adt.data) <- paste0(colnames(adt.data), "-1")
    # assay object stores info about RNA/Cite/etc assays
    adt_assay <- CreateAssayObject(counts = adt.data)
    seurat_obj[["ADT"]] <- adt_assay
  }
  seurats[[sample_names[s1]]] <- seurat_obj
}
```

##################################### QUALITY CONTROL -(dead cells, low coverage cells, etc) ############################

```{r QC}
for (s1 in sample_names) {
  # load object to do QC for
  seurat_obj <- seurats[[s1]]
  # determine the mitochondrial dna amount// add to metadata using [] operator
  seurat_obj[["percent.mt"]] <- PercentageFeatureSet(seurat_obj, pattern = mtPattern)
  # log scale the counts for features and all RNA// assign and access metadata values using $ operator
  seurat_obj$log10GenesPerUMI <- log10(seurat_obj$nFeature_RNA) / log10(seurat_obj$nCount_RNA)
  # append to list the updated metadata containing object
  seurats[[s1]] <- seurat_obj
}
```

# Looping over QC plots
1:09
```{r qcPlots, include=TRUE, fig.width=15}
violin_list <- list()
dotplot_list <- list()
metadata_list <- list()
for (v1 in names(seurats)) {
  cat("### QC Plots ", v1, "\n")
  vPlot <- VlnPlot(seurats[[v1]],
    features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,
    log = TRUE
  ) + NoLegend()
  ggsave(filename = file.path(out_path, paste0("qc_violin_plot_", v1, ".png")))
  violin_list[[v1]] <- vPlot
  metadata <- seurats[[v1]]@meta.data
  metadata_list[[v1]] <- metadata
  d1 <- metadata %>%
    ggplot(aes(x = nCount_RNA, y = nFeature_RNA, color = percent.mt)) +
    geom_point() +
    ggtitle("RNA1") +
    scale_colour_gradient(low = "gray90", high = "black") +
    stat_smooth(method = lm) +
    scale_x_log10() +
    scale_y_log10() +
    theme_classic() +
    geom_vline(xintercept = 500) +
    geom_hline(yintercept = 250)
  ggsave(filename = file.path(out_path, paste0("genes_UMI_corr_plot_", v1, ".png")))
  dotplot_list[[v1]] <- d1
  cat("\n")
  print(vPlot | d1)
  cat("\n")
}
```

# Merging data batches without integration: (4 mice, adding names for each sample in the meta-data)

```{r merge_objects}
seurat_obj <- merge(x = seurats[[1]], y = seurats[2:length(seurats)], add.cell.id = sample_names)
# Add sample names to metadata:
# Pull out seurat_obj's metadata as a dataframe
metadata <- seurat_obj@meta.data
# Add cell names as column
metadata$"cells" <- rownames(metadata)
# Add sample names to metadata
metadata$sample <- NA
for (s in sample_names) {
  nameStr <- paste0("^", s)
  metadata$sample[which(str_detect(metadata$cells, nameStr))] <- s
}
# replace metadata
seurat_obj@meta.data <- metadata
```

# Pre-filtering cellcount (store cell-numbers before and after)

```{r orig_counts}
orig_idents <- table(seurat_obj$orig.ident)
knitr::kable(orig_idents, caption = "Pre-Filtering Cell Counts")
```

# Filtering cells by thresholding 

```{r filtering}
nCount_cutoff <- 10 # min counts per cell
nFeature_cutoff <- 100 # min features present per cell
genesPerUMI_cutoff <- 0.5 # min ratio of features by counts (log scaled) per cell
pctMt_cutoff <- 10 # max mt% allowed per cell
cellsPerGene_cutoff <- 10 # min cells one gene should have non-zero expression in

# Filter out low quality reads using selected thresholds - these will change with experiment
filtered_seurat <- subset(
  x = seurat_obj,
  subset = (nCount_RNA >= nCount_cutoff) &
    (nFeature_RNA >= nFeature_cutoff) &
    (log10GenesPerUMI > genesPerUMI_cutoff) &
    (percent.mt < pctMt_cutoff)
)

# Filter out features with too few cells per gene

# as samples were merged into one seurat object, they were kept in different layers of the RNA assay
# join the cells from the samples into one layer (without cell-matching/integration of modalities/or sample assay data)
filtered_seurat <- JoinLayers(filtered_seurat)
counts <- GetAssayData(object = filtered_seurat, slot = "counts")
# Output a logical vector for every gene on whether more than zero counts per cell
nonzero <- counts > 0
# Sums all TRUE values and returns TRUE if more than <cutoff #> TRUE values per gene
keep_genes <- Matrix::rowSums(nonzero) >= cellsPerGene_cutoff
# Only keeping those genes expressed in more than <cutoff #> cells
filtered_counts <- counts[keep_genes, ]

# Reassign to filtered Seurat object
filtered_seurat <- CreateSeuratObject(filtered_counts, meta.data = filtered_seurat@meta.data)
filtered_table <- table(filtered_seurat$orig.ident)
```

# Cutoffs

nCount minimum: `r nCount_cutoff`

nFeature minimum: `r nFeature_cutoff`

log10 genes per UMI minimum: `r genesPerUMI_cutoff`

Percent MT DNA maximum: `r pctMt_cutoff`

Minimum cells per gene: `r cellsPerGene_cutoff`

# Post-filtering cell counts

```{r filtering2}
# saveRDS(filtered_seurat, file = paste0(out_path, "filtered_seurat_object.rds"))
knitr::kable(filtered_table, caption = "Filtered Cell Counts")
```

# Saving post-filtering object with ALL SAMPLES

``` {r loading filtered seurat object}
# filtered_seurat <- readRDS(paste0(out_path, "Agingfiltered_seurat_object.rds"))
```

# Post-filtering QC plots 

```{r filtering3}
Idents(filtered_seurat) <- filtered_seurat$orig.ident
VlnPlot(filtered_seurat,
  features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,
  log = TRUE, pt.size = 0
) + NoLegend()
ggsave(filename = file.path(out_path, "filtered_violin_plot_all.png"))


metadata <- filtered_seurat@meta.data
metadata %>%
  ggplot(aes(x = nCount_RNA, y = nFeature_RNA, color = percent.mt)) +
  geom_point() +
  scale_colour_gradient(low = "gray90", high = "black") +
  stat_smooth(method = lm) +
  scale_x_log10() +
  scale_y_log10() +
  theme_classic() +
  geom_vline(xintercept = 500) +
  geom_hline(yintercept = 250)
ggsave(filename = file.path(out_path, "filtered_genes_UMI_corr_plot.png"))
```

############################################## CELL EMBEDDINGS #####################################################

```{r normalization and pca}
# default norm is LogNorm (~~FPKM)// use SCTransform (~~TPM)
norm_seurat <- NormalizeData(filtered_seurat)
# The function FindVariableFeatures identifies the most highly variable genes (default 2000 genes) by fitting a line to the relationship of log(variance) and log(mean) using loss smoothing,
# uses this information to standardize the data, then calculates the variance of the standardized data.
# This helps avoid selecting genes that only appear variable due to their expression level.
norm_seurat <- FindVariableFeatures(norm_seurat, selection.method = "mean.var.plot")
norm_seurat <- ScaleData(norm_seurat, features = VariableFeatures(norm_seurat))
pca_seurat <- RunPCA(norm_seurat)
# elbow Plot (PCA loadings)
ElbowPlot(pca_seurat, ndims = 20, reduction = "pca")
```

# Integration with Harmony (label the integrated sample by batch/sample_name in the umap to quantify batch effect)

```{r integration}
# batch effect correction post data-integration
integ_seurat <- RunHarmony(pca_seurat, group.by.vars = "sample")
# plot umap and label by sample_name (M4/M11/F5/F17)
## ////// Integration was not needed, so decided to use merged data for further analysis
```

# Clustering cells in the latent space 

```{r clustering1}
# Set number of dimensions & resolution values to be used in clustering
nDims <- 18
cluster_resolution <- 0.8
```

# Params for clustering and running umap

Number of dimensions: `r nDims`
Cluster resolution: `r cluster_resolution`

```{r clustering2}
# KNN graph using PC feature's euclidean distance in latent space
cluster_seurat <- FindNeighbors(integ_seurat, reduction = "pca", dims = 1:nDims)
# Louvain clustering (modularity optimisation)
cluster_seurat <- FindClusters(cluster_seurat, resolution = cluster_resolution, verbose = FALSE)
cluster_seurat <- RunUMAP(cluster_seurat, dims = 1:nDims)
DimPlot(cluster_seurat, label = TRUE)
ggplot2::ggsave(file.path(out_path, "clusters_all.png"))
```

``` {r}
# saveRDS(cluster_seurat, file = paste0(out_path, "cluster_seurat.rds"))
cluster_seurat <- readRDS("/ix/djishnu/Zarifeh/ML_MM/Aditi/result/Aging/Agingcluster_seurat.rds")
```

########################## DEG ANALYSIS : PER CLUSTER (CLUSTER SPECIFIC BIOMARKERS) #########################################

```{r degs}
aging.markers <- FindAllMarkers(cluster_seurat, only.pos = TRUE)
aging.markers %>%
  group_by(cluster) %>%
  dplyr::filter(avg_log2FC > 1)
write.xlsx(x = aging.markers, file = file.path(out_path, "all_markers.xlsx"))
```
``` {r up and downregulated}
aging.markers_updown <- FindAllMarkers(cluster_seurat, only.pos = FALSE)
aging.markers_updown %>%
  group_by(cluster) %>%
  dplyr::filter(avg_log2FC > 1)
write.xlsx(x = aging.markers_updown, file = file.path(out_path, "all_markers_updown.xlsx"))
```
``` {r cluster 0}
cluster0.markers <- FindMarkers(cluster_seurat, ident.1 = 0, logfc.threshold = 1, test.use = "roc", only.pos = FALSE)
head(cluster0.markers)
```
``` {r cluster 1}
cluster1.markers <- FindMarkers(cluster_seurat, ident.1 = 1, logfc.threshold = 1, test.use = "roc", only.pos = FALSE)
head(cluster1.markers)
```
``` {r cluster 2}
cluster2.markers <- FindMarkers(cluster_seurat, ident.1 = 2, logfc.threshold = 1, test.use = "roc", only.pos = FALSE)
head(cluster2.markers)
```

``` {r cluster 3}
cluster3.markers <- FindMarkers(cluster_seurat, ident.1 = 3, logfc.threshold = 1, test.use = "roc", only.pos = FALSE)
head(cluster3.markers)
```

``` {r cluster 4}
cluster4.markers <- FindMarkers(cluster_seurat, ident.1 = 4, logfc.threshold = 1, test.use = "roc", only.pos = FALSE)
head(cluster4.markers)
```

``` {r Assigning cell type identity to clusters }
aging.markers <- read.xlsx("/ix/djishnu/Zarifeh/ML_MM/Aditi/result/Aging/all_markers_updown.xlsx")

aging.markers %>%
  group_by(cluster) %>%
  dplyr::filter(avg_log2FC > 1) %>%
  slice_head(n = 5) %>%
  ungroup() -> top5
DoHeatmap(cluster_seurat, features = top5$gene) + NoLegend()
ggsave("/ix/djishnu/Zarifeh/ML_MM/Aditi/result/Aging/cluster_heatmap_top5.png", device = "png", width = 20, height = 30)

top5$gene
DotPlot(
  object = cluster_seurat,
  features = unique(top5$gene),
  cluster.idents = TRUE,
  scale.by = "radius",
  cols = c("lightgrey", "darkgreen"),
) + RotatedAxis()

ggsave("/ix/djishnu/Zarifeh/ML_MM/Aditi/result/Aging/DOTPLOT.png", device = "png", width = 18, height = 15)
```

``` {r Feature plots }
# senescence markers:
FeaturePlot(cluster_seurat, features = c("Cdkn1a", "Myh8", "Col19a1", "Lrrk2", "Acta2"), ncol = 3)
# ggplot2::ggsave(file.path(out_path, "feature_plot_all_1.png"))
```
``` {r making reference}
cell_annot <- data.frame(
  Cell_type = c(
    "MuSC", "MF-I", "MF-IIB", "MF-IIX", "MF-IIA",
    "NMJ", "MTJ", "MF-other", "FB", "Tenocyte",
    "mSchwann", "nmSchwann", "SMC", "Pericyte",
    "CapEC", "ArtEC", "VenEC", "LymphEC", "M2_LYVE1",
    "Macrophages_LAM", "Monocyte", "Neutrophil",
    "B cell", "Adipocyte", "RBC"
  ),
  Markers = c(
    "Pax7", "Lpl, Tnnt1, Tnnc1, Tnnt3, Tnnc2",
    "Tnnt3, Tnnc2", "Myh1, Tnnc2, Tnnt3",
    "Lpl, Myh2, Tnnc2, Tnnt3", "Tnnc2, Tnnt3",
    "Col22a1, Tnnc2, Tnnt3, Rgcc",
    "Tnnc2, Tnnt3, Rgcc", "Lpl, Col1a1, Dcn, Pdgfra",
    "Lpl, Acta2, Tnmd, Col1a1, Dcn",
    "Mpz, Mbp, Rgcc", "Ncam1, Mbp, Rgcc",
    "Acta2, Myh11", "Acta2, Kcnj8, Rgs5, Pecam1, Cdh5, Rgcc, Lpl",
    "Pecam1, Cdh5, Rgcc, Lpl", "Pecam1, Cdh5, Rgcc",
    "Pecam1, Cdh5", "Ccl21a, Mmm1, Rgs5",
    "Lyve1, F13a1, Fcgr3, Clec10a, Cd14",
    "Fcgr3, Spp1, F13a1, Lyve1", "Fcgr3, Cd14",
    "S100a8, Fcgr3, Cd14", "Cd79a",
    "Cyp2e1, Lpl, Tnnt3", "Hba-a1, Lpl, Rgcc, Cdh5, Rgcc, Pecam1"
  ),
  stringsAsFactors = FALSE
)

# Write the data frame to a CSV file
write.csv(cell_annot, file = "/ix/djishnu/Zarifeh/ML_MM/Aditi/result/cell_annot.csv", row.names = FALSE)
```

``` {r cluster annotation}
aging.markers_updown <- FindAllMarkers(cluster_seurat, only.pos = FALSE)
```

``` {r cluster annotation}
print(cell_annot)
```

### Heatmap showing top 5 DE genes in each cluster

```{r heatmap, eval=FALSE}
# Load the CSV file containing cell types and markers
cell_annot <- read.csv("/ix/djishnu/Zarifeh/ML_MM/Aditi/result/cell_annot.csv", stringsAsFactors = FALSE)
aging.markers <- read.xlsx("/ix/djishnu/Zarifeh/ML_MM/Aditi/result/Aging/all_markers_updown.xlsx")
# allmarkers %>%
# group_by(cluster) %>%
# top_n(n = 10, wt = avg_log2FC) -> top10
# DoHeatmap(cluster_seurat, features = top5$gene) + NoLegend()

aging.markers %>%
  group_by(cluster) %>%
  dplyr::filter(avg_log2FC > 1) %>%
  slice_head(n = 20) %>%
  ungroup() -> top20

current_markers <- unlist(strsplit(cell_annot[, "Markers"], ", "))
shared <- intersect(top20$gene, current_markers)
top20_shared <- top20[top20$gene %in% shared, ]
## Cluster identification
FeaturePlot(cluster_seurat, features = shared[1:4], ncol = 2)
ggsave(file.path(out_path, "markers_1.png"))
FeaturePlot(cluster_seurat, features = shared[5:8], ncol = 2)
ggsave(file.path(out_path, "markers_2.png"))
FeaturePlot(cluster_seurat, features = shared[9:12], ncol = 2)
ggsave(file.path(out_path, "markers_3.png"))
FeaturePlot(cluster_seurat, features = shared[13], ncol = 2)
ggsave(file.path(out_path, "markers_4.png"))

# top20_shared$cell_type <- NA
# top20_shared$cell_type[1] <- "MF-IIX"
# top20_shared$cell_type[2] <- "FB, Tenocyte"
# top20_shared$cell_type[3] <- "CapEC, ArtEC , VenEC"
# top20_shared$cell_type[4] <- "MTJ"
# top20_shared$cell_type[5] <- "CapEC, ArtEC, RBC"
# top20_shared$cell_type[6] <- "Pericyte"
# top20_shared$cell_type[7] <- "MuSC"
# top20_shared$cell_type[8] <- "nmSchwann"
# top20_shared$cell_type[9] <- "nmSchwann"
# top20_shared$cell_type[10] <- "M2_LYVE1, Macrophages_LAM"
# top20_shared$cell_type[11] <- "MF-I, MF-IIA, CapEC, RBC"
# top20_shared$cell_type[12] <- "MF-I, MF-IIB, MF-IIX, MF-IIA, NMJ, MTJ, MF-other"
# top20_shared$cell_type[13] <- "MF-IIA"
# top20_shared$cell_type[14] <- "MF-I, MF-IIA, CapEC, RBC"
# top20_shared$cell_type[15] <- "MF-I, MF-IIB, MF-IIX, MF-IIA, NMJ, MTJ, MF-other"
# top20_shared$cell_type[16] <- "MF-I, MF-IIB, MF-IIX, MF-IIA, NMJ, MTJ, MF-other"
```

# Reclustering (reducing cluster numbers)

## Gene marker visualiztion
## Renaming clusters based on scores

```{r TcellsOnly_genetic_markers_rename}
new.cluster.ids <- c("0", "1", "MF-IIX", "3", "4", "MF-IIX", "6", "7", "MF-IIA", "9", "FB,Tenocyte", "CapEC,ArtEC,VenEC", "MTJ", "Pericyte", "MuSC", "nmSchwann", "Neuronmuscular Junction", "17")
names(new.cluster.ids) <- levels(cluster_seurat)
# rename cluster identities by cell type name
cluster_seurat <- RenameIdents(cluster_seurat, new.cluster.ids)
DimPlot(cluster_seurat, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
ggsave(file.path(out_path, "clusters_annot.png"))
```
``` {r more markers from the paper}
FeaturePlot(cluster_seurat, features = c("Xist", "Fhl1", "Pdk4", "Fat3", "Myh1"), ncol = 3)
```
``` {r cluster identification using all the markers from the paper}
Markers <- c(
  "Pax7", "Lpl", "Tnnt1", "Tnnc1", "Tnnt3", "Tnnc2",
  "Tnnt3", "Tnnc2", "Myh1", "Tnnc2", "Tnnt3",
  "Lpl", "Myh2", "Tnnc2", "Tnnt3", "Tnnc2", "Tnnt3",
  "Col22a1", "Tnnc2", "Tnnt3", "Rgcc",
  "Tnnc2", "Tnnt3", "Rgcc", "Lpl", "Col1a1", "Dcn", "Pdgfra",
  "Lpl", "Acta2", "Tnmd", "Col1a1", "Dcn",
  "Mpz", "Mbp", "Rgcc", "Ncam1", "Mbp", "Rgcc",
  "Acta2", "Myh11", "Acta2", "Kcnj8", "Rgs5", "Pecam1", "Cdh5", "Rgcc", "Lpl",
  "Pecam1", "Cdh5", "Rgcc", "Lpl", "Pecam1", "Cdh5", "Rgcc",
  "Pecam1", "Cdh5", "Ccl21a", "Mmm1", "Rgs5",
  "Lyve1", "F13a1", "Fcgr3", "Clec10a", "Cd14",
  "Fcgr3", "Spp1", "F13a1", "Lyve1", "Fcgr3", "Cd14",
  "S100a8", "Fcgr3", "Cd14", "Cd79a",
  "Cyp2e1", "Lpl", "Tnnt3", "Hba-a1", "Lpl", "Rgcc", "Cdh5", "Rgcc", "Pecam1"
)
# FeaturePlot(cluster_seurat, features = Markers[1:4], ncol = 2)
VlnPlot(cluster_seurat,
  features = unique(Markers), assay = "RNA",
  group.by = "seurat_clusters",
  stack = TRUE, flip = TRUE
)
ggsave(file.path(out_path, "markers_paper.png"))
```

``` {r cluster identification using all immune cell markers}
marker_genes <-
  c(
    "Itgam", # Cd11b - Myeloid marker (mouse)
    "Adgre1", # F4/80,
    "Pax5", # B
    "Ighd", "Cd27", # Naive (mouse) B cell markers (IgD+, CD27-)
    "Cd3d", "Cd3e", "Cd8a", # T
    "Klrb1c", "Prf1", "Klrk1", "Gzma", "Gzmb", # NK
    "Itga2", "Ncam1", # NK-T
    "Cd83", # DCs
    "Cd14", "Cd68", # Macs - note that Cd16 never comes up
    "Itgax", # DCs
    "Ly6c1",
    "Cd74", # MHC-II mouse marker (used by Renthal 2022 to identify immune cells in TG)
    "Ptgs2", "Irf5", "Nos2", # Mouse M1 Mac Markers
    "Stat1", "Retnla", # Mouse M1 Mac Markers (less helpful)
    "Il12a", "Il23a", "Cd163", # M1 vs M2 (M1: IL-12 and IL23 high with CD163 neg and M2 the opposite)
    "Cd163", # M2
    "Arg1", # M2a
    "Socs3", "Cd86", # M2b
    "Ccr2", "Slamf6", # M2c
    "Tlr1", "Tlr8", "Scarb1", # M2c (less helpful)
    "Vegfa", # M2d,
    "Cx3cr1" # Tissue-res Mac
  )

VlnPlot(cluster_seurat,
  features = marker_genes, assay = "RNA",
  group.by = "seurat_clusters",
  stack = TRUE, flip = TRUE
)
ggsave(file.path(out_path, "immune_markers_vlnplot.png"))

FeaturePlot(cluster_seurat, features = "Pdk4")
```

```{r TcellsOnly_genetic_markers_rename after immune markers}
new.cluster.ids <- c("0", "1", "MF-IIX", "3", "4", "MF-IIX", "6", "7", "MF-IIA", "9", "FB,Tenocyte", "CapEC,ArtEC,VenEC", "MTJ", "Pericyte", "MuSC", "nmSchwann", "Neuronmuscular Junction", "F4/80, M2")
names(new.cluster.ids) <- levels(cluster_seurat)
cluster_seurat <- RenameIdents(cluster_seurat, new.cluster.ids)
DimPlot(cluster_seurat, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
ggsave(file.path(out_path, "clusters_annot.png"))

FeaturePlot(cluster_seurat, features = "Pdk4")
```

```{r markers from https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9792217/ }
FeaturePlot(cluster_seurat, features = c("Csf1r", "Nkg7", "Cd3e", "Pdgfra", "Pecam1", "Pax7", "Cd34", "Tnnt1", "Tnn"), ncol = 3)
ggsave(file.path(out_path, "clusters_annot_PMC9792217markers.png"))
# LRRK2, CDKN1A, MYH8, COL19A1
FeaturePlot(cluster_seurat, features = c("Kdr"), ncol = 3)
```
```{r markers in http://cloud.capitalbiotech.com/SingleCellBase/search_result.jsp}
FeaturePlot(cluster_seurat, features = c("Ccl17", "Ccl22", "Ccr7", "Pecam1", "lgbp7", "Ctla2a", "Col4a1"), ncol = 3)
```

``` {r cluster identification using all immune cell markers}
marker_genes <-
  c(
    "Itgam", # Cd11b - Myeloid marker (mouse)
    "Adgre1", # F4/80,
    "Pax5", # B
    "Ighd", "Cd27", # Naive (mouse) B cell markers (IgD+, CD27-)
    "Cd3d", "Cd3e", "Cd8a", # T
    "Klrb1c", "Prf1", "Klrk1", "Gzma", "Gzmb", # NK
    "Itga2", "Ncam1", # NK-T
    "Cd83", # DCs
    "Cd14", "Cd68", # Macs - note that Cd16 never comes up
    "Itgax", # DCs
    "Ly6c1",
    "Cd74", # MHC-II mouse marker (used by Renthal 2022 to identify immune cells in TG)
    "Ptgs2", "Irf5", "Nos2", # Mouse M1 Mac Markers
    "Stat1", "Retnla", # Mouse M1 Mac Markers (less helpful)
    "Il12a", "Il23a", "Cd163", # M1 vs M2 (M1: IL-12 and IL23 high with CD163 neg and M2 the opposite)
    "Cd163", # M2
    "Arg1", # M2a
    "Socs3", "Cd86", # M2b
    "Ccr2", "Slamf6", # M2c
    "Tlr1", "Tlr8", "Scarb1", # M2c (less helpful)
    "Vegfa", # M2d,
    "Cx3cr1" # Tissue-res Mac
  )

VlnPlot(cluster_seurat,
  features = marker_genes, assay = "RNA",
  group.by = "seurat_clusters",
  stack = TRUE, flip = TRUE
)
ggsave(file.path(out_path, "immune_markers_vlnplot.png"))

FeaturePlot(cluster_seurat, features = "Flnc")
```

```{r from new re Aditi has sent}
# marker_genes <- c("Smc4", "Cdk1", "Hmgb2", "Ctsa", "Cstb", "Ctsz", "Ccl9", "Mrc1", "C1qa",
#                 "Cxcr1", "Cd74", "Cd83", "Fscn1", "Flt3", "Xcr1", "Cd72", "Cd209a", "Cxcr4",
#                "S100a8", "S100a9", "Cd8a", "Cd3a", "Cd8b1", "Cd4", "Nkg4", "Klre1", "Klrd1",
#               "Pecam1", "Cdh5", "Acta2", "Pdgfra", "Col3a1", "Acta1", "Sdc1", "Pax7"
#              )

# marker_genes <- c("Cxcl10", "Ctsl", "Fabp5", "Ly6c2", "Ccr2", "Ccl6", "Tgfbi", "Ccl2", "Cd68", "Ccl8", "Lyz2",
#                 "Ltgam", "Cd163", "Mapk14", "Fcgr3", "Csf1r", "Cxcl16", "Ctss", "Cd200r1", "H2-Ab1", "H2-Aa",
#                "ltgax", "Csf1", "Sell", "Cd14", "Mmp9", "Ptprc", "Ccl5", "Ccr7", "Cd3e", "Foxp3", "Cxcr3", "Cd27",
#               "Tbx21", "Gata3", "Gzma", "Klra4", "Klra7", "Klra8", "Klre1"
# )
# marker_genes <- c("Klrd1", "H2-Oa", "H2-Ob", "Lgkc", "Ms4a1", "Cd19", "Cd22", "Hba-a1", "Hba-a2", "Hbb-bs", "Hbb-bt"
# )
marker_genes <- c(
  "Cd55", "Pi16", "Igfbp5", "Dpp4", "Ly6a", "Hif1a", "Lpl", "Fbln5", "Cxcl12", "Thy1", "Tnfaip6", "Cd164",
  "Cd34", "Il33", "Hmgb2", "Cdk1", "Egfr",
  "Smc4", "Mcm5", "Tyms", "Rrm2", "Dio2", "Rspo3"
)
# marker_genes <- c("Cox6a2", "Cox6c", "Atp5e", "Atp5g1", "Ckm", "Chrne")
marker_genes <- c("Ttn", "Myh4", "Myh1", "Chrne", "Col22a1", "Pax7", "Titin", "Nrap", "Myh9", "Myh10", "Enah", "Finc", "Fhod3")
# marker_genes <- c("Cox6a2", "Cox6c", "Atp5e", "Atp5g1", "Ckm", "Chrne")
marker_genes <- c("Xist", "Fhl1", "Pdk4", "Fat3", "Myh1")
# marker_genes <- c( "Hic1", "Adam12", "Bmp5", "Myoc", "Col3a1", "Bmp1", "Gsn", "Pdgfra", "Bgn", "Hdlbp", "Cxcl5", "Cxcl1", "Ccl7", #"Mmp14", "Ctsk", "Col1a1", "Dcn", "Mmp2", "Apod", "Tnmd", "Scx", "Ptn", "Mpz")
# marker_genes <- c("Pax7", "Sdc1", "Acta1", "Myh1", "Myh4", "Acta2", "Apold1", "Ednrb", "Rgs5", "Myl9", "Myh11", "Cspg4",
# "Pdgfrb", "Mcam", "Alpl", "Hey1", "Cdh5", "Pecam1", "Cxcl12",
# "Ly6a", "Kdr", "Lpl", "Cd34", "Vwf", "Hif1a", "Icam1", "Lrg1", "Aplnr", "Apln", "S100a9", "S100a8", "Csf1", "Cxcr4", "Itgam", "Smc4", "Cdk1", "Hmgb2"
# )
VlnPlot(cluster_seurat,
  features = marker_genes, assay = "RNA",
  group.by = "seurat_clusters",
  stack = TRUE, flip = TRUE
)
ggsave(file.path(out_path, "AditiNewref_markers_vlnplot6.png"))
```

# Order of cells in seurat object
# F7 : [ ,1:10169]
# F15 : [ ,10170:19875]
# M4 : [ ,19876:31893]
# M11 : [ ,31894:40106]